addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1471,8 +1472,8 @@ export class Wysiwyg extends Component {
                     this.odooEditor.historyUnpauseSteps();
                     this.odooEditor.historyStep();
                     const link = data.linkDialog.$link[0];
-                    setSelection(link, 0, link, link.childNodes.length, false);
-                    link.focus();
+                    const linkIndex = childNodeIndex(link);
+                    setSelection(link.parentElement, linkIndex+1, link.parentElement, linkIndex+1, false);
```

Comment:
I'm curious. What happens if you remove the two lines above? Isn't the selection at the place we want already?

Reply:
No, the selection isn't correct; without those lines, the selection will appear at the beginning of the editable area instead of after the link.